flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/FlinkMatchers.java

```diff
@@ -35,7 +35,14 @@
 import java.util.function.Function;
 import java.util.function.Predicate;
 
-/** Some reusable hamcrest matchers for Flink. */
+/**
+ * Some reusable hamcrest matchers for Flink.
+ *
+ * @deprecated You should assertj assertions, which have built-in assertions for {@link
+ *     CompletableFuture}. To check chains of {@link Throwable} causes, use {@link
+ *     FlinkAssertions#assertThatChainOfCauses(Throwable)}
+ */
+@Deprecated
```

Comment:
Changes for the core Flink testing infrastructure should not happen in a `table-common` commit. Please create a separate commit.

Reply:
Refactored in a separate commit annotated as hotfix